cli/internal/state/state.go

```diff
@@ -7,23 +7,126 @@ SPDX-License-Identifier: AGPL-3.0-only
 )
 
+// ReadFromFile reads the state file at the given path and returns the state.
+func ReadFromFile(fileHandler file.Handler, path string) (*State, error) {
+	state := &State{}
+	if err := fileHandler.ReadYAML(path, &state); err != nil {
+		return nil, fmt.Errorf("reading state file: %w", err)
+	}
+	return state, nil
+}
+
 // State describe the entire state to describe a Constellation cluster.
 type State struct {
 	Version        string         `yaml:"version"`
 	Infrastructure Infrastructure `yaml:"infrastructure"`
+	ClusterValues  ClusterValues  `yaml:"clusterValues"`
 }
 
-// NewState creates a new state with the given infrastructure.
-func NewState(Infrastructure Infrastructure) State {
-	return State{
-		Version:        Version1,
-		Infrastructure: Infrastructure,
+// New creates a new cluster state (file).
+func New() *State {
+	return &State{
+		Version: Version1,
+	}
+}
+
+// NewFromIDFile creates a new cluster state file from the given ID file.
+func NewFromIDFile(idFile clusterid.File) *State {
+	s := New()
+
+	if idFile.ClusterID != "" {
+		s.ClusterValues.ClusterID = idFile.ClusterID
+	}
+
+	if idFile.OwnerID != "" {
+		s.ClusterValues.OwnerID = idFile.OwnerID
+	}
+
+	if idFile.UID != "" {
+		s.Infrastructure.UID = idFile.UID
+	}
+
+	if idFile.IP != "" {
+		s.Infrastructure.ClusterEndpoint = idFile.IP
+	}
+
+	if len(idFile.APIServerCertSANs) > 0 {
+		s.Infrastructure.APIServerCertSANs = idFile.APIServerCertSANs
+	}
+
+	if idFile.InitSecret != nil {
+		s.Infrastructure.InitSecret = string(idFile.InitSecret)
 	}
+
+	if idFile.AttestationURL != "" {
+		s.Infrastructure.Azure = &Azure{
+			AttestationURL: idFile.AttestationURL,
+		}
+	}
+
+	if len(idFile.MeasurementSalt) > 0 {
+		s.ClusterValues.MeasurementSalt = idFile.MeasurementSalt
+	}
+
+	return s
+}
+
+// SetInfrastructure sets the infrastructure state.
+func (s *State) SetInfrastructure(infrastructure Infrastructure) *State {
+	s.Infrastructure = infrastructure
+	return s
+}
+
+// SetClusterValues sets the cluster values.
+func (s *State) SetClusterValues(clusterValues ClusterValues) *State {
+	s.ClusterValues = clusterValues
+	return s
+}
+
+// WriteToFile writes the state to the given path, overwriting any existing file.
+func (s *State) WriteToFile(fileHandler file.Handler, path string) error {
+	if err := fileHandler.WriteYAML(path, s, file.OptMkdirAll, file.OptOverwrite); err != nil {
+		return fmt.Errorf("writing state file: %w", err)
+	}
+	return nil
+}
+
+// Merge merges the state information from other into the current state.
+// If a field is set in both states, the value of the other state is used.
+func (s *State) Merge(other *State) (*State, error) {
+	if err := mergo.Merge(s, other, mergo.WithOverride); err != nil {
+		return nil, fmt.Errorf("merging state file: %w", err)
+	}
+	return s, nil
+}
+
+// ClusterName returns the name of the cluster.
+func (s *State) ClusterName(cfg *config.Config) string {
+	return cfg.Name + "-" + s.Infrastructure.UID
+}
```

Comment:
We could just make this a field in the state file to avoid having to keep this function

Reply:
Can I just edit this in the RFC?